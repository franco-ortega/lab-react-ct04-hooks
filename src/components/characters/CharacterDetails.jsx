import React from 'react';
import PropTypes from 'prop-types';

export default function CharacterDetails({
  id,
  name,
  japaneseName,
  image,
  quote,
  personality,
  species,
  skill,
  goal,
  roast,
  milk,
  sugar
})
{
  return (
    <>
        Character Details
      <h2>{id}</h2>
      <h2>{name}</h2>
      <h2>{japaneseName}</h2>
      <h2><img src={image} alt={name} style={{ height: '100px' }} /></h2>
      <h2>{quote}</h2>
      <h2>{personality}</h2>
      <h2>{species}</h2>
      <h2>{skill}</h2>
      <h2>{goal}</h2>
      <h2>{roast}</h2>
      <h2>{milk}</h2>
      <h2>{sugar}</h2>
    </>
  );
}

CharacterDetails.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  japaneseName: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  quote: PropTypes.string.isRequired,
  personality: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired,
  skill: PropTypes.string.isRequired,
  goal: PropTypes.string.isRequired,
  roast: PropTypes.string.isRequired,
  milk: PropTypes.string.isRequired,
  sugar: PropTypes.string.isRequired
};
