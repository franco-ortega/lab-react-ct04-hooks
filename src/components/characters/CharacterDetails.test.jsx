import React from 'react';
import { render } from '@testing-library/react';
import CharacterDetails from './CharacterDetails';

describe('CharacterDetails component', () => {
  it('renders CharacterDetails', () => {
    const { asFragment } = render(<CharacterDetails
      name="words"
      japaneseName="words"
      image="words"
      quote="words"
      personality="words"
      species="words"
      skill="words"
      goal="words"
      // coffee={{ object: 'words' }}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
