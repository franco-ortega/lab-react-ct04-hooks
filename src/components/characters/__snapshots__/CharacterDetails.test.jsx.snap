// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetails component renders CharacterDetails 1`] = `
<DocumentFragment>
  Character Details
  <h2>
    words
  </h2>
  <h2>
    words
  </h2>
  <img
    alt="words"
    src="words"
    style="height: 100px;"
  />
  <p>
    words
  </p>
  <p>
    words
  </p>
  <p>
    words
  </p>
  <p>
    words
  </p>
  <p>
    words
  </p>
</DocumentFragment>
`;
